From 25e81c4080c863fff516946dd087516b1c934ab4 Mon Sep 17 00:00:00 2001
From: Charles Steinkuehler <charles@steinkuehler.net>
Date: Sat, 6 Dec 2014 20:05:33 +0000
Subject: [PATCH 40/44] cape: LCD4: Fix GPIO buttons Correct errant GPIO
 settings in exclusive-use and pinmux portion of device tree overlay: gpio0_3
 incorrectly listed gpio3_16 & gpio0_15 were missing

Signed-off-by: Len Shelton <len@probotix.com>
[charles@steinkuehler.net: whitespace cleanup]
[charles@steinkuehler.net: remove PBX from compatible = ]
Signed-off-by: Charles Steinkuehler <charles@steinkuehler.net>
---
 firmware/capes/BB-BONE-LCD4-01-00A1.dts | 94 ++++++++++++++++++---------------
 1 file changed, 50 insertions(+), 44 deletions(-)

diff --git a/firmware/capes/BB-BONE-LCD4-01-00A1.dts b/firmware/capes/BB-BONE-LCD4-01-00A1.dts
index 5c72f1e..84a520d 100644
--- a/firmware/capes/BB-BONE-LCD4-01-00A1.dts
+++ b/firmware/capes/BB-BONE-LCD4-01-00A1.dts
@@ -5,6 +5,9 @@
  * it under the terms of the GNU General Public License version 2 as
  * published by the Free Software Foundation.
  */
+
+/* fixed incorrect pinouts 12-1-14 Len Shelton */
+
 /dts-v1/;
 /plugin/;
 
@@ -18,42 +21,44 @@
 	/* state the resources this cape uses */
 	exclusive-use =
 		/* the pin header uses */
-		"P8.45",	/* lcd: lcd_data0 */
-		"P8.46",	/* lcd: lcd_data1 */
-		"P8.43",	/* lcd: lcd_data2 */
-		"P8.44",	/* lcd: lcd_data3 */
-		"P8.41",	/* lcd: lcd_data4 */
-		"P8.42",	/* lcd: lcd_data5 */
-		"P8.39",	/* lcd: lcd_data6 */
-		"P8.40",	/* lcd: lcd_data7 */
-		"P8.37",	/* lcd: lcd_data8 */
-		"P8.38",	/* lcd: lcd_data9 */
-		"P8.36",	/* lcd: lcd_data10 */
-		"P8.34",	/* lcd: lcd_data11 */
-		"P8.35",	/* lcd: lcd_data12 */
-		"P8.33",	/* lcd: lcd_data13 */
-		"P8.31",	/* lcd: lcd_data14 */
-		"P8.32",	/* lcd: lcd_data15 */
-		"P8.27",	/* lcd: lcd_vsync */
-		"P8.29",	/* lcd: lcd_hsync */
-		"P8.28",	/* lcd: lcd_pclk */
-		"P8.30",	/* lcd: lcd_ac_bias_en */
-		"P9.27",	/* lcd: gpio3_19 */
-		"P9.12",	/* led: gpio1_28 */
-		"P9.14",	/* pwm: ehrpwm1a */
-		"P9.15",	/* keys: gpio1_16 */
-		"P9.23",	/* keys: gpio1_17 */
-		"P9.16",	/* keys: gpio1_19 */
-		"P9.21",	/* keys: gpio0_3 */
-		/* the hardware IP uses */
-		"gpio3_19",
-		"gpio1_28",
-		"gpio1_16",
-		"gpio1_17",
-		"gpio1_19",
-		"gpio0_3",
+		"P8.45",	/* lcd: lcd_data0		*/
+		"P8.46",	/* lcd: lcd_data1		*/
+		"P8.43",	/* lcd: lcd_data2		*/
+		"P8.44",	/* lcd: lcd_data3		*/
+		"P8.41",	/* lcd: lcd_data4		*/
+		"P8.42",	/* lcd: lcd_data5		*/
+		"P8.39",	/* lcd: lcd_data6		*/
+		"P8.40",	/* lcd: lcd_data7		*/
+		"P8.37",	/* lcd: lcd_data8		*/
+		"P8.38",	/* lcd: lcd_data9		*/
+		"P8.36",	/* lcd: lcd_data10		*/
+		"P8.34",	/* lcd: lcd_data11		*/
+		"P8.35",	/* lcd: lcd_data12		*/
+		"P8.33",	/* lcd: lcd_data13		*/
+		"P8.31",	/* lcd: lcd_data14		*/
+		"P8.32",	/* lcd: lcd_data15		*/
+		"P8.27",	/* lcd: lcd_vsync		*/
+		"P8.29",	/* lcd: lcd_hsync		*/
+		"P8.28",	/* lcd: lcd_pclk		*/
+		"P8.30",	/* lcd: lcd_ac_bias_en		*/
+		"P9.27",	/* lcd: gpio3_19 *	DISPEN	*/
+		"P9.12",	/* led: gpio1_28	LED	*/
+		"P9.14",	/* pwm: ehrpwm1a	PWM_BL	*/
+		"P9.15",	/* keys: gpio1_16	LEFT	*/
+		"P9.23",	/* keys: gpio1_17	RIGHT	*/
+		"P9.16",	/* keys: gpio1_19	UP	*/
+		"P9.30",	/* keys: gpio3_16	DOWN	*/
+		"P9.24",	/* keys: gpio0_15	ENTER	*/
+
+		"gpio3_19",	/*	DISPEN			*/
+		"gpio1_28",	/*	LED			*/
+		"gpio1_16",	/*	LEFT			*/
+		"gpio1_17",	/*	RIGHT			*/
+		"gpio1_19",	/*	UP			*/
+		"gpio3_16",	/*	DOWN			*/
+		"gpio0_15",	/*	ENTER			*/
 		"lcd",
-		"ehrpwm1a";
+		"ehrpwm1a";	/*	PWM_BL			*/
 
 	fragment@0 {
 		target = <&am33xx_pinmux>;
@@ -99,10 +104,11 @@
 
 			bone_lcd4_cape_keys_00A2_pins: pinmux_bone_lcd4_cape_keys_00A2_pins {
 				pinctrl-single,pins = <
-					0x040 0x2f	/* gpmc_a0.gpio1_16, INPUT | PULLDIS | MODE7 */
-					0x044 0x2f	/* gpmc_a1.gpio1_17, INPUT | PULLDIS | MODE7 */
-					0x04c 0x2f 	/* gpmc_a3.gpio1_19, INPUT | PULLDIS | MODE7 */
-					0x154 0x2f	/* P9_21 spi0_d0.gpio0_3 INPUT | PULLDIS | MODE7 */
+					0x040 0x27	/* gpio1_16	INPUT | MODE7	LEFT	*/
+					0x044 0x27	/* gpio1_17  	INPUT | MODE7	RIGHT	*/
+					0x04C 0x27	/* gpio1_19  	INPUT | MODE7	UP	*/
+					0x198 0x27	/* gpio3_16	INPUT | MODE7	DOWN	*/
+					0x184 0x27	/* gpio0_15	INPUT | MODE7	ENTER	*/
 				>;
 			};
 
@@ -192,7 +198,7 @@
 					debounce_interval = <50>;
 					linux,code = <105>;
 					label = "left";
-					gpios = <&gpio2 16 0x1>;
+					gpios = <&gpio2 16 0x1>;	/* LEFT		*/
 					gpio-key,wakeup;
 					autorepeat;
 				};
@@ -200,7 +206,7 @@
 					debounce_interval = <50>;
 					linux,code = <106>;
 					label = "right";
-					gpios = <&gpio2 17 0x1>;
+					gpios = <&gpio2 17 0x1>;	/* RIGHT	*/
 					gpio-key,wakeup;
 					autorepeat;
 				};
@@ -208,7 +214,7 @@
 					debounce_interval = <50>;
 					linux,code = <103>;
 					label = "up";
-					gpios = <&gpio2 19 0x1>;
+					gpios = <&gpio2 19 0x1>;	/* UP		*/
 					gpio-key,wakeup;
 					autorepeat;
 				};
@@ -216,7 +222,7 @@
 					debounce_interval = <50>;
 					linux,code = <108>;
 					label = "down";
-					gpios = <&gpio4 16 0x1>;
+					gpios = <&gpio4 16 0x1>;	/* DOWN		*/
 					gpio-key,wakeup;
 					autorepeat;
 				};
@@ -224,7 +230,7 @@
 					debounce_interval = <50>;
 					linux,code = <28>;
 					label = "enter";
-					gpios = <&gpio1 15 0x1>;
+					gpios = <&gpio1 15 0x1>;	/* ENTER	*/
 					gpio-key,wakeup;
 				};
 			};
-- 
2.4.6

